public/js/src/module/form.js

```diff
@@ -419,5 +421,49 @@ Form.prototype.updateValidityInUi = function (control, result) {
     }
 };
 
+Form.prototype.goToTarget = function (target, options = {}) {
+    if (target) {
+        if (this.pages.active && !options.isPageFlip) {
+            // Flip to page
+            this.pages.flipToPageContaining($(target));
+        }
+        // check if the target has a form control
+        if (target.closest('.calculation, .setvalue, .setgeopoint')) {
+            // It is up to the apps to decide what to do with this event.
+            target.dispatchEvent(events.GoToInvisible());
+        }
+        // check if the nearest question or group is irrelevant after page flip
+        if (target.closest('.or-branch.disabled')) {
+            // It is up to the apps to decide what to do with this event.
+            target.dispatchEvent(events.GoToIrrelevant());
+        }
+
+        // Focus on the first non .ignore form control which is not currently readonly.
+        // If the element is hidden (e.g. because it's been replaced by a widget),
+        // the focus event will not fire, so we also trigger an applyfocus event that widgets can listen for.
+        let selector =
+            'input:not(.ignore):not([readonly]), textarea:not(.ignore):not([readonly]), select:not(.ignore):not([readonly])';
+
+        // For repeat DOM, prevent focus on DN dome when all element is readonly or ignore(#733)
+        if (target.querySelector('.repeat-number')) {
```

Comment:
I think you may have to separate 2 selector options: (1) `target.closest('.question')` with the unchanged selector and (2) `else` excluding discrepancy notes. This will also fix a possible yet-undiscovered bug I mention later. So (2) would cover the cases where target is a form or repeat else (or page, or whatever else that is not a question).

I wish there was a more elegant solution, but I think you're right that overwriting goToTarget seems best (because comment widgets are not officially supported in enketo/enketo). The problem will be to keep this function up-to-date with future upstream changes. However, at least once we migrate to OpenClinica/enketo-oc (the monorepo) we could make these changes directly in enketo-core which will be helpful for things like this.

Reply:
@theywa  did you take a look at the comments provided by @MartijnR

Reply:
sorry @kkrumlian not yet, I plan to focus on it tomorrow

@MartijnR 
I think you are right about the goToTarget, but since it's on enketo-core, I tried to prevent making a change there afraid to make another issue. But I tried to make an update on goToTarget in enketo-core by adding another parameter to indicate isClone or not and decided not to use it.

I'll check the dn first on Dom, and for the API, I did check gotoQuery works fine with the current update. 
I'll also test your suggestion and will inform you later.
Thank you.

Reply:
@theywa, I think the query parameter is not correct, so Enketo assumes the question does not exist. Here's an example I use to get the API to return a well-formed (escaped etc) query parameter (for another form):

```bash
curl --user something: -d "server_url=http://localhost:8005&form_id=Max&instance=\
<max1 xmlns:oc= \"http://openclinica.org/xforms\" oc:complete= \"true\">\
    <try1>\
        <hh_num>3</hh_num>\
        <hh_num_comment></hh_num_comment>\
    </try1>\
</max1>&instance_id=a&go_to=/max1/try1/hh_num_comment&ecid=a" \
http://localhost:8005/oc/api/v1/instance/edit
```

Reply:
I didn't test it myself though...